src/modules/commander/Arming/PreFlightCheck/checks/sdcardCheck.cpp

```diff
@@ -50,12 +50,17 @@ bool PreFlightCheck::sdcardCheck(orb_advert_t *mavlink_log_pub, const bool repor
 	param_get(param_find("COM_ARM_SDCARD"), &param_com_arm_sdcard);
 
 	if (param_com_arm_sdcard > 0) {
-		bool sdcard_present{false};
+		static bool sdcard_present{false};
```

Comment:
Can you do this without static variables?

How about adding something like in `vehicle_status_flag` for the sdcard that latches true on first success? Then check it here before calling. https://github.com/PX4/PX4-Autopilot/blob/0f29b871010cfa1b42991f5a885bbceade1b6cdc/src/modules/commander/Arming/PreFlightCheck/PreFlightCheck.cpp#L67

Reply:
I just changed the implementation.